plugin/src/main/java/io/jenkins/plugins/analysis/core/portlets/PullRequestMonitoringPortlet.java

```diff
@@ -0,0 +1,130 @@
+package io.jenkins.plugins.analysis.core.portlets;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import com.google.gson.JsonObject;
+
+import edu.hm.hafner.analysis.Severity;
+
+import org.kohsuke.stapler.bind.JavaScriptMethod;
+import hudson.Extension;
+import hudson.model.Run;
+
+import io.jenkins.plugins.analysis.core.model.AnalysisResult;
+import io.jenkins.plugins.analysis.core.model.ResultAction;
+import io.jenkins.plugins.monitoring.MonitorPortlet;
+import io.jenkins.plugins.monitoring.MonitorPortletFactory;
+
+/**
+ * A portlet that can be used for the pull-request-monitoring dashboard
+ * (https://github.com/jenkinsci/pull-request-monitoring-plugin).
+ * It renders an interactive sunburst diagram for each {@link ResultAction},
+ * which is registered at the current users {@link Run}.
+ *
+ * @author Simon Symhoven
+ */
+public class PullRequestMonitoringPortlet implements MonitorPortlet {
+
+    private final ResultAction action;
+
+    /**
+     * Creates a new {@link PullRequestMonitoringPortlet}.
+     *
+     * @param action
+     *              the corresponding {@link ResultAction}.
+     */
+    public PullRequestMonitoringPortlet(final ResultAction action) {
+        this.action = action;
+    }
+
+    @Override
+    public String getTitle() {
+        return action.getLabelProvider().getName();
+    }
+
+    @Override
+    @JavaScriptMethod
+    public String getId() {
+        return action.getResult().getId();
+    }
+
+    @Override
+    public int getPreferredWidth() {
+        return 350;
+    }
+
+    @Override
+    public int getPreferredHeight() {
+        return 350;
+    }
+
+    @Override
+    public String getIconUrl() {
+        return action.getIconFileName();
+    }
+
+    @Override
+    public Optional<String> getDetailViewUrl() {
+        return Optional.of("../" + action.getUrlName());
```

Comment:
I find the relative URL difficult to understand but it seems to work as intended.

This is called from [portlet.jelly](https://github.com/jenkinsci/pull-request-monitoring-plugin/blob/38077ebb55e65859a0e34183c4b0057764d0ea1d/src/main/resources/portlet/portlet.jelly#L25), which is referenced from [MonitoringDefaultAction/index.jelly](https://github.com/jenkinsci/pull-request-monitoring-plugin/blob/6e0e418a008f149fad739ec097bf8617ea3faff1/src/main/resources/io/jenkins/plugins/monitoring/MonitoringDefaultAction/index.jelly#L36). MonitoringDefaultAction is attached to a Run and [its getUrlName method](https://github.com/jenkinsci/pull-request-monitoring-plugin/blob/c1a90464474738e6b45d02c2272a631eae37c135/src/main/java/io/jenkins/plugins/monitoring/MonitoringDefaultAction.java#L57-L60) returns "pull-request-monitoring". So the ".." walks back to the Run and then you get to the ResultAction, which is also attached to the same Run.

Nothing to fix in this PR I think, but documenting the base URL in [MonitorPortlet.getDetailViewUrl](https://github.com/jenkinsci/pull-request-monitoring-plugin/blob/c1a90464474738e6b45d02c2272a631eae37c135/src/main/java/io/jenkins/plugins/monitoring/MonitorPortlet.java#L56-L63) might be useful, especially because the result of MonitorPortlet.getIconUrl is interpreted differently.

Reply:
Thats true. It is probably easier and more understandable if I change this to an absolute url so that would be `Optional.of(action.getAbsolutUrl())`. I will expand the documentation of the method, that there is no base url, do you agree?

Reply:
In the old wiki, [Hyperlinks in HTML](https://wiki.jenkins.io/display/JENKINS/Hyperlinks+in+HTML) recommends not using absolute URLs that include the scheme and host. Absolute paths are OK, though. I don't know whether the advice there is still valid.

Reply:
iconUrl and detailViewUrl using both the app relative url. As soon as I released it, I will update this PR. Thanks.

Reply:
It is still valid to always use relative paths (and never use absolute paths). Otherwise using a reverse proxy will be quite difficult. What makes more sense here: let the pr-view plugin handle the navigation from the dashboard. A portlet should only define the portion of the path that starts after the given build URL. So here:

```suggestion
        return Optional.of(action.getUrlName());
```